lib/elixir/lib/macro.ex

```diff
@@ -276,13 +276,21 @@ defmodule Macro do
     {call, line, List.insert_at([], integer, expr)}
   end
 
-  def pipe(expr, {call, line, args} = call_args, integer) when is_list(args) do
-    if is_atom(call) and Identifier.binary_op(call) != :error do
-      raise ArgumentError,
-            "cannot pipe #{to_string(expr)} into #{to_string(call_args)}, " <>
-              "the #{to_string(call)} operator can only take two arguments"
-    else
-      {call, line, List.insert_at(args, integer, expr)}
+  def pipe(expr, {op, line, args} = op_args, integer) when is_list(args) do
+    cond do
+      is_atom(op) and Identifier.unary_op(op) != :error and length(args) == 1 ->
+        raise ArgumentError,
+              "cannot pipe #{to_string(expr)} into #{to_string(op_args)}, " <>
+                "the #{to_string(op)} operator can only take two arguments. " <>
```

Comment:
Unary operators can only take one argument. Here is my suggestion:

```elixir
def pipe(_expr, {op, _line, [arg]}, _integer) when op == :+ or op == :- do
  raise ArgumentError,
        "piping into a unary operator is not supported, please use the qualified name: " <>
          "Kernel.#{op}(#{to_string(arg)}), instead of #{op}#{to_string(arg)}"
end

def pipe(expr, {op, line, args} = op_args, integer) when is_list(args) do
  cond do
    is_atom(op) and Identifier.unary_op(op) != :error ->
      raise ArgumentError,
            "cannot pipe #{to_string(expr)} into #{to_string(op_args)}, " <>
              "the #{to_string(op)} operator can only take one argument"

    is_atom(op) and Identifier.binary_op(op) != :error ->
      raise ArgumentError,
            "cannot pipe #{to_string(expr)} into #{to_string(op_args)}, " <>
              "the #{to_string(op)} operator can only take two arguments"

    true ->
      {op, line, List.insert_at(args, integer, expr)}
  end
end
```

Reply:
Ah, that makes more sense!